drbert/data/dataset_readers.py

```diff
@@ -0,0 +1,416 @@
+from torchtext import data
+from torchtext import datasets
+
+
+class DatasetReader(object):
+    """The parent class for all dataset readers. A user is mean to interact with a subclass of this
+    class (e.g. DocumentClassificationDatasetReader), not this class itself.
+
+    Args:
+        path (str): Common prefix of the splits’ file paths.
+        partitions (dict): A dictionary keyed by partition ('train', 'test', 'validation')
+            containing the suffix to add to `path` for that partition.
+        tokenizer (transformers.PreTrainedTokenizer): The function used to tokenize strings into
+            sequential examples.
+        format (str): The format of the data file. One of “CSV”, “TSV”, or “JSON” (case-insensitive).
+        skip_header (bool) : Optional, whether to skip the first line of the input file. Defaults to
+            False.
+        batch_sizes (tuple): Optional, tuple of batch sizes to use for the different splits, or None
+            to use the same batch_size for all splits. Defaults to None.
+        lower: (bool): Optional, True if text should be lower cased, False if not. Defaults to
+            False.
+        sort_key (function): A key to use for sorting examples in order to batch together examples
+            with similar lengths and minimize padding. The sort_key provided to the Iterator
+            constructor overrides the sort_key attribute of the Dataset, or defers to it if None.
+
+    Raises:
+        ValueError: If batch_sizes is not a tuple or None.
+        ValueError: If batch_sizes is not None and `len(batch_sizes) != len(partitions)`.
+    """
+    def __init__(self, path, partitions, tokenizer, format, skip_header=False, batch_sizes=None,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'format'](https://app.codacy.com/app/JohnGiorgi/drbert/pullRequest?prid=4302662)